subprojects/docs/src/docs/userguide/dependencyLocking.adoc

```diff
@@ -0,0 +1,144 @@
+</sample>
+++++
+
+=== Running a build with a lockfile present
+
+The moment a build needs to resolve a configuration that has locking enabled and it finds a matching lockfile,
+it will use it to verify that the given configuration still resolves the same versions.
+The validation is as follows:
+
+* Existing entries in the lockfile must be matched in the build
+** A version mismatch or missing resolved module causes a build failure
+* Resolution result must not contain extra dependencies compared to the lock file content
+
+=== Generating and updating lockfiles
+
+In order to generate or update a lockfile, you specify the `--write-locks` command line argument in addition to the normal tasks that would trigger configurations to be resolved.
+This will cause the creation of a lockfile for each resolved configuration in that build execution.
+Note that if a lockfile existed previously, it is overwritten.
+
+==== Lock all configurations in one build execution
+
+When locking multiple configurations, you may want to lock them all at once, during a single build execution.
+
+For this, you have two options:
+
+* Run `./gradlew dependencies --write-locks`.
+This will effectively lock all resolvable configurations that have locking enabled.
+Note that in a multi project setup, `dependencies` only is executed on _one_ project, the root one in this case.
+* Declare a custom task that will resolve all configurations
+++++
+<sample id="locking-all" dir="userguide/dependencyManagement/dependencyLocking/lockingAllConfigurations" title="Resolving all configurations">
+    <sourcefile file="build.gradle" snippet="resolve-all"/>
+</sample>
+++++
+
+=== Controlled lock file update
+
+In order to update only specific modules of a configuration, you can use the `--update-locks` command line flag.
+It takes a comma (`,`) separated list of module notations.
+In this mode, the existing lock file is still used as input to resolution, filtering out the modules targeted by the update.
+Note however that the resolution may cause other module versions to update, as dictated by the Gradle resolution rules.
+
+----
+❯ ./gradlew classes --update-locks org:foo,org:bar
+----
+
+Wildcards, indicated with `*`, can be used in the group or module name. They can be the only character or appear at the end of the group or module respectively.
+The following wildcard notations are valid:
+
+* `org*:foo`
+* `org:*`
+* `*:bar`
+* `org*:bar*`
+
+=== Deleting a lockfile
+
+Simply remove the file matching the configurations where you no longer want locking.
+
+=== Locking limitations
+
+It is currently not possible to lock the `classpath` configuration used for build plugins.
+
+=== Nebula locking plugin
+
+This feature is inspired by the https://github.com/nebula-plugins/gradle-dependency-lock-plugin[Nebula Gradle dependency lock plugin].
+It is effectively an adaptation of the features of that plugin that leverage the recent developments around dependency management in Gradle.
```

Comment:
Let's avoid mentioning recency in the user guide. It will become outdated before we notice.

Reply:
It seems we have 2 camps there. I told @ljacomet to remove the mention to Nebula, when @eriwen asked him to integrate back. I tend to think like you Piotr that it's too simple to get outdated with an explicit mention. If we care about migration, a blog post or a guide would be better suited I think.

Reply:
This is a valid concern especially given that the core capability will essentially supersede the Nebula plugin. Perhaps the Nebula plugin documentation would be a better place to document the migration and we could add a link here? What do you think @eriwen and @ljacomet?

Reply:
The important thing is that we give those using the Nebula plugins a migration resource. Nebula plugin documentation would be an appropriate place, but I think it's on us to work with @DanielThomas and others to ensure it's gets there if we all agree.

Reply:
I rephrased this to still have the "inspired by" reference, but then only send you there if you are migrating.